sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryBlobClientTests.java

```diff
@@ -362,20 +357,19 @@ public void uploadFromStream() throws IOException {
     @Test
     public void downloadToFile() throws IOException, InterruptedException, ExecutionException {
         File output = File.createTempFile("temp", "in");
-
         ContainerRegistryBlobAsyncClient asyncClient = createAsyncClient(createDownloadBlobClient(SMALL_CONTENT, SMALL_CONTENT_SHA256));
-        try (AsynchronousFileChannel outputChannel = AsynchronousFileChannel
-            .open(output.toPath(), StandardOpenOption.CREATE, StandardOpenOption.WRITE, StandardOpenOption.READ)) {
+        try (FileOutputStream outputStream = new FileOutputStream(output)) {
             StepVerifier.create(asyncClient.downloadStream(SMALL_CONTENT_SHA256)
-                    .flatMap(result -> result.writeValueToAsync(IOUtils.toAsynchronousByteChannel(outputChannel, 0))))
+                    .flatMap(result -> FluxUtil.writeToOutputStream(result.toFluxByteBuffer(), outputStream)))
                 .verifyComplete();
 
-            ByteBuffer result = ByteBuffer.wrap(SMALL_CONTENT.toBytes());
-            outputChannel.read(result, 0).get();
-
-            // make Java 8 happy
-            Buffer flip = result.flip();
-            assertArrayEquals(SMALL_CONTENT.toBytes(), (byte[]) flip.array());
+            outputStream.flush();
+            try (FileInputStream inputStream = new FileInputStream(output)) {
```

Comment:
I'd rather use `Files.readAllBytes` (https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#readAllBytes-java.nio.file.Path-) in this test rather than what's being done now as this reading could be flaky if we can't read all bytes in one go for some reason.